src/ds5/ds5-options.cpp

```diff
@@ -700,20 +705,26 @@ namespace librealsense
         return *_range;
     }
 
-    auto_gain_limit_option::auto_gain_limit_option(hw_monitor& hwm, sensor_base* ep, option_range range)
-        : option_base(range), _hwm(hwm), _sensor(ep)
+    auto_gain_limit_option::auto_gain_limit_option(hw_monitor& hwm, sensor_base* ep, option_range range, limits_option* gain_limit_enable)
+        : option_base(range), _hwm(hwm), _sensor(ep), _gain_limit_enable(gain_limit_enable)
     {
         _range = [range]()
         {
             return range;
         };
+        set(get_range().max);
     }
 
     void auto_gain_limit_option::set(float value)
     {
         if (!is_valid(value))
             throw invalid_value_exception("set(enable_auto_gain) failed! Invalid Auto-Gain mode request " + std::to_string(value));
 
+        _gain_limit_enable->set_cached_limit(value);
+        if(value != get_range().max && _gain_limit_enable->query() == 0.f)
```

Comment:
As long as the value is valid the toggle shall be set on.
At this point the valid is already verified to be within the range - I don't think the check is required